fab/core.py

```diff
@@ -74,8 +80,9 @@ def fab_forward_kl(self, batch_size: int) -> torch.Tensor:
         x_ais, log_w_ais = self.annealed_importance_sampler.sample_and_log_weights(batch_size)
         x_ais = x_ais.detach()
         log_w_ais = log_w_ais.detach()
+        w_ais = torch.softmax(log_w_ais, dim=0)
         log_q_x = self.flow.log_prob(x_ais)
-        return - torch.mean(torch.exp(log_w_ais) * log_q_x)
+        return - torch.mean(w_ais * log_q_x)
```

Comment:
This is not related to the buffer MR, but I am including it from my dev branch as it improved the stability of the forward kl loss.